artemis-server/src/test/java/org/apache/activemq/artemis/core/config/impl/FileConfigurationParserTest.java

```diff
@@ -257,6 +257,26 @@ public void testParsingDefaultServerConfigWithENCMaskedPwd() throws Exception {
       assertEquals("helloworld", bconfig.getPassword());
    }
 
+   @Test
+   public void testParsingOverflowPageSize() throws Exception {
+      FileConfigurationParser parser = new FileConfigurationParser();
```

Comment:
This test, test the new change, but doesnt test for what we are protecting from. E.g whats the underlying issue being fixed

Reply:
E.g. other sizes are quite happily long currently also. Why is this special

Reply:
> E.g. other sizes are quite happily long currently also. Why is this special

I see in Page::write()/Page::read() all operations are limited to int range, such as PagingStoreImpl::currentPageSize, Page::size, and local variables like fileSize, processedBytes in Page::readFromSequentialFile(), etc.

> This test, test the new change, but doesnt test for what we are protecting from. E.g whats the underlying issue being fixed

I'm not sure what you mean. We're protecting page-size-bytes from being greater than Integer.MAX_VALUE(2147483647). So if we set page-size-bytes to 2147483648, the broker is expected to fail fast to throw exception , correct?

Reply:
Point more so was i assume the same issue may lay in other persistence such as journal etc